src/bl/userFacade.ts

```diff
@@ -98,4 +110,55 @@ export class UserFacade {
             throw new BadRequest(e)
         }
     }
+
+    static async reset(email: string): Promise<User> {
+        if (!email) {
+            throw new BadRequest('This user does not exist')
+        }
+        try {
+            const user = await UserFacade.getByEmail(email)
+            const token = uuid('photon.igpolytech.fr', uuid.DNS)
+            user.resetToken = token
+            user.resetTimeStamp = new Date()
+            console.log(user.resetTimeStamp)
+
+            sendMail(email, resetPassword(user.username, user.id, token))
+            
+            return await getRepository(User).save(user)
```

Comment:
Maybe save the user first then send the mail.

Reply:
ok